### PR TITLE
Array buffer asset item

### PR DESCRIPTION
**Pull request by takahirox**
**Description:**

This PR adds `responseType` attribute to `a-asset-item`.
It's necessary for `THREE.GLTFLoader` in r84. #2378
And it'll be helpful for user components which need `arraybuffer` data.

It has a limitation.
If we attempt to load the same data as different `responseType`s,
only the `responseType` of the first attempt is applied
because it loads from cache  if exists.

I think it won't cause any problems because
there's no situation that we wanna load the same data with different `responseType`s.
(If exists, let me know.)

**Changes proposed:**
- Add `responseType` attribute to `a-asset-item`. 
- Add `responseType` tests

### CODE REVIEW

File: src/core/a-assets.js [hunk 1]
```diff
@@ -98,6 +98,9 @@ registerElement('a-asset-item', {
       value: function () {
         var self = this;
         var src = this.getAttribute('src');
+        var responseType = this.getAttribute('responseType');
+        if (responseType === null) responseType = 'text';
```
Comment (ngokevin) [inline src/core/a-assets.js]: `var responseType = this.getAttribute('response-type') || 'text';`

html attribute case insensitive

Comment (takahirox) [inline src/core/a-assets.js]: Oops, thanks! I'll update.

**Comment by ngokevin** [comment]
I've added docs and made the quick update. Thanks!

**Comment by takahirox** [comment]
Thanks for the update!

I have a question.

You changed the order of

    THREE.Cache.remove(XHR_SRC);
    var assetItem = document.createElement('a-asset-item');

Is there any special reasons?
And if so, we should change the order here, too?

https://github.com/aframevr/aframe/blob/3c074bdde332f3d28649eb9bbf3c3e207107fe9d/tests/core/a-assets.test.js#L276

**Pull request merged**